src/parser/deserializer.ts

```diff
@@ -45,6 +45,8 @@ export interface DeserializeOptions {
   index?: number;
 
   raw?: boolean;
+  /** allows for opt-in utf-8 validation */
+  validation?: Document;
```

Comment:
```suggestion
  validation?: { utf8: Record<string, boolean> };
```
Isn't it a nested utf8 key that points to the key->boolean? 
Just to add a bit stricter typing.